kubernetes/utils/create_from_yaml.py

```diff
@@ -0,0 +1,74 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from os import path
+import sys
+
+from six import iteritems
+
+import yaml
+
+from kubernetes import client
+
+def create_from_yaml(yaml_file, verbose=0, **kwargs):
+    """
+    Perform an action from a yaml file. Pass 1 for verbose to
+    print confirmation information.
+
+    Available parameters for generating the client:
+    :param configuration: Configuration for the client.
+    :param host: The base path for the server to call.
+    :param header_name: a header to pass when making calls to the API.
+    :param header_value: a header value to pass when making calls to the API.
+
+    Available parameters for performing the subsequent action:
+    :param async_req bool
+    :param bool include_uninitialized: If true, partially initialized resources are included in the response.
+    :param str pretty: If 'true', then the output is pretty printed.
+    :param str dry_run: When present, indicates that modifications should not be persisted. An invalid or unrecognized dryRun directive will result in an error response and no further processing of the request. Valid values are: - All: all dry run stages will be processed
+    """
+    
+    client_params = ["configuration", "host", "header_name", "header_value"]
+    action_params = ["async_req", "include_unitialized", "pretty", "dry_run",
+        '_return_http_data_only', '_preload_content', '_request_timeout']
+    client_args = {}
+    action_args = {}
+    params = locals()
+    for key, val in iteritems(params["kwargs"]):
+        if key in client_params:
+            client_args[key] = val
+        elif key in action_params:
+            action_args[key] = val
+
+    k8s_client = client.api_client.ApiClient(**client_args)
+    with open(path.abspath(yaml_file)) as f:
+        dep = yaml.load(f)
+        api_type, _, api_version = dep["apiVersion"].partition("/")
+        if api_version == "":
+                api_version = api_type
+                api_type = "core"
+        fcn_to_call = "{0}{1}Api".format(api_type.capitalize(),
+                api_version.capitalize())
+        k8s_api = getattr(client, fcn_to_call)(k8s_client)
+        action_type = dep["kind"]
+        if "namespace" in dep["metadata"]:
+                dep_namespace = dep["metadata"]["namespace"]
+        else:
+                dep_namespace = "default"
+        resp = getattr(k8s_api, "create_namespaced_{0}".format(action_type.lower()))(
```

Comment:
We also need to be able to create cluster-scoped resources (e.g. create a namespace)